build/lib/builder.js

```diff
@@ -28,6 +38,47 @@ function thisOS() {
 	return osName;
 }
 
+function determineBabelOptions(platform) {
```

Comment:
Ok, so you're now generating platform-specific versions of the ti.main.js. I definitely is tricky in the sense that technically act platform will get a different copy of the shared code, but obviously means it only gets what it needs (and we trust babel to do the right thing). I think it's a pretty sound approach so long as it doesn't bloat the SDK size (God, I hope the shared code isn't that big!)

But, I would definitely rewrite this SDK build method here still - you're grabbing the android and iOS data you need and then basically deleting the other platform after the fact. The easiest "fix" would be to just construct all of the object we need and then have a switch or if/else blocks to add in the platform-specific `targets` key/value pair(s).

In fact, given that we'd have differing values between platforms, it may make sense to have the platforms code grab the data it needs and return it (i.e. have the IOS, Android and Windows classes have a shared method called babelTarget() or something that returns an Object we can inject into the common babel options object we're creating, like: `{ iOS: minSupportedIosSdk }`.

I get that I'm being super-picky here...

 If you go up the call stack here you're calling #transpile() right before creating an instance of the Platform class and calling build() on it anyways. maybe just construct the object and pass it in to the transpile method here, or come up with a shared base class for the Platform builder classes that holds the common transpile logic and have the subclasses override some method to return the relevant `targets` object.